src/Sylius/Behat/Context/Api/Shop/AddressContext.php

```diff
@@ -163,6 +172,116 @@ public function iSetTheAddressOfAsDefault(string $fullName): void
         $this->customerClient->update();
     }
 
+    /**
+     * @When /^I try to edit the (address of "([^"]+)")$/
+     */
+    public function iTryToEditTheAddressOf(AddressInterface $address)
+    {
+        $this->addressClient->buildUpdateRequest((string) $address->getId());
+    }
+
+    /**
+     * @When I change the first name to :firstName
+     */
+    public function iChangeTheFirstNameTo(string $firstName)
+    {
+        $this->addressClient->addRequestData('firstName', $firstName);
+    }
+
+    /**
+     * @When I change the last name to :lastName
+     */
+    public function iChangeTheLastNameTo(string $lastName)
+    {
+        $this->addressClient->addRequestData('lastName', $lastName);
+    }
+
+    /**
+     * @When I change the street to :street
+     */
+    public function iChangeTheStreetTo(string $street)
+    {
+        $this->addressClient->addRequestData('street', $street);
+    }
+
+    /**
+     * @When I change the city to :city
+     */
+    public function iChangeTheCityTo(string $city)
+    {
+        $this->addressClient->addRequestData('city', $city);
+    }
+
+    /**
+     * @When I change the postcode to :postcode
+     */
+    public function iChangeThePostcodeTo(string $postcode)
+    {
+        $this->addressClient->addRequestData('postcode', $postcode);
+    }
+
+    /**
+     * @When I choose :provinceName as my province
+     */
+    public function iChooseAsMyProvince(string $provinceName)
+    {
+        /** @var ProvinceInterface */
+        $province = $this->provinceProvider->findByName($provinceName);
```

Comment:
instead of this provider you should use here transformer (`src/Sylius/Behat/Context/Transform/ProvinceContext.php`)